lib/stores/S3Store.js

```diff
@@ -437,139 +474,107 @@ class S3Store extends DataStore {
             });
     }
 
-    write(req, file_id, offset) {
-        return new Promise((resolve, reject) => {
-            this._getMetadata(file_id)
-                .then((metadata) => this._countParts(file_id).then((part_number) => ([part_number, metadata])))
-                .then((results) => {
-                    const [part_number, metadata] = results;
-                    const current_part_number = part_number + 1;
-                    const total_parts = Math.ceil(
-                        parseInt(metadata.file.upload_length, 10) / this.part_size
-                    );
-
-                    // s3 sdk requires a content length
-                    // even though it is a stream (?)
-                    if (current_part_number === total_parts) {
-                        req.length = metadata.file.upload_length - offset;
-                    }
-                    else {
-                        req.length = this.part_size;
-                    }
-
-                    // we need to know if the connection was closed
-                    // to gracefully handle s3 sdk errors
-                    let isClosed = false;
-                    req.on('close', () => {
-                        isClosed = true;
-                    });
-
-                    return this._processUpload(metadata, req, current_part_number)
-                        .then(resolve)
-                        .catch((err) => {
-                            // we'll get in here when the client closes/cancels
-                            // the request however s3 is expecting more data since
-                            // we had to set a content length in the incoming stream
-                            if (['RequestTimeout', 'NoSuchUpload'].includes(err.code) && isClosed) {
-                                if (err.code === 'RequestTimeout') {
-                                    console.warn('[S3Store] Request "close" event was emitted,'
-                                        + 'however S3 was expecting more data. Failing gracefully...');
-                                }
-
-                                if (err.code === 'NoSuchUpload') {
-                                    console.warn('[S3Store] Request "close" event was emitted,'
-                                        + 'however S3 was expecting more data. Most likely the'
-                                        + ' upload is already finished/aborted. Failing gracefully...');
-                                }
-
-                                return this.getOffset(file_id, true)
-                                    .then((current_offset) => resolve(current_offset.size))
-                                    .catch(reject);
+    write(req, file_id) {
+        return this._getMetadata(file_id)
+            .then((metadata) =>
+                this._countParts(file_id).then((part_number) => [part_number, metadata])
+            )
+            .then(async(results) => {
```

Comment:
Can we have a space here or does the linter not allow this:

> async (results) => {

Reply:
linter is not happy indeed. But that will change in the rewrite